mitosheet/src/mito/utils/actions.tsx

```diff
@@ -714,6 +784,58 @@ export const getActions = (
             searchTerms: ['filter', 'remove', 'delete'],
             tooltip: "Filter this dataframe based on the data in a column."
         },
+
+        [ActionEnum.FilterToCellValue]: {
+            type: 'build-time',
+            staticType: ActionEnum.FilterToCellValue,
+            titleContextMenu: 'Filter to Cell Value',
+            iconContextMenu: FilterIcon,
+            longTitle: 'Filter column',
+            actionFunction: async () => {
+                // We turn off editing mode, if it is on
+                setEditorState(undefined);
+                const {startingColumnIndex} = gridState.selections[0];
+                const columnID = sheetData.data[startingColumnIndex].columnID;
+                const { columnFilters, cellValue, columnDtype } = getCellDataFromCellIndexes(sheetData, gridState.selections[0].startingRowIndex, gridState.selections[0].startingColumnIndex);
+                const filters = columnFilters?.filters ?? [];
```

Comment:
A lot of this data (e.g. the columnID) exists at the top of the action function, already accessible. Let's remove any duplicated variable definitions, and then just move the rest of it up (e.g. filters we can get up top) so we can use them in any action that needs them.

Reply:
Since you disable this action when there is more than one cell selected, this should be safe to use the variables defined at the top of this function.